framework/kms-keyring.md

```diff
@@ -266,9 +266,12 @@ in the input encrypted data key list with the following conditions, until it suc
 To attempt to decrypt a particular [encrypted data key](#structures.md#encrypted-data-key),
 OnDecrypt MUST call [KMS Decrypt](#kms-decrypt).
 
-The [KMS client](#kms-client) that calls [KMS Decrypt](#kms-decrypt) MUST be the
-client returned by the [client supplier](#client-supplier).
-The client MUST be a client that calls the AWS region specified in the [generator](#generator) ARN.
+
+If an AWS region can be extracted from the [encrypted data key](structures.md#encrypted-data-key)'s
```

Comment:
An AWS region MUST be extractable from the EDK. This is because the EDK key provider info serializes the keyARN, and never the alias. If the KMS Keyring cannot extract the region in this case, it should result in an error.